crates/counter-agg/src/stable.rs

```diff
@@ -0,0 +1,62 @@
+//! These data structures are stable, meaning they may not change even in
+//! their layout in memory, as the raw bytes-in-memory are serialized and
+//! exchanged by PostgreSQL.
+//!
+//! Note that [MetricSummary] is already in violation, as it does not lock in
+//! a memory representation and the Rust project makes no guarantees to
+//! preserve this across releases of the compiler.  We should bump its
+//! serialization version and repr(C) the new one.
+
+use serde::{Deserialize, Serialize};
+
+use stats_agg::stats2d::StatsSummary2D;
+use time_series::TSPoint;
+
+use crate::range::I64Range;
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+// TODO Our one serialization test (counter_byte_io) passes, but is that just luck?
+//#[repr(C)]
+pub struct MetricSummary {
```

Comment:
I don't love creating this new struct that only differs by `bounds` being an option.  Is this because you thought the original `MetricSummary` wasn't something that could be changed?  In this case, I'd prefer changing the original object.  Failing that, maybe we could make `bounds` non-public and add a `bounds()` accessor that returns an option?